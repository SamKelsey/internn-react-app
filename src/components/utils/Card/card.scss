@import "../../styleVariables";

.about-card {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  font-family: "Montserrat", sans-serif;
  text-align: center;
  color: white;
  border-radius: 10px;
  transition: all 0.3s;
  padding: 15px 15px;
  margin: 10px;
  width: 500px;

  &:hover {
    box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;

    i {
      transform: scale(1.2);
    }
  }

  i {
    font-size: 2.5rem;
    margin: 0 auto;
    padding: 0;
    transition: all 0.3s ease;
  }

  h2 {
    font-size: 1.5rem;
    font-weight: 600;
    margin: 0;
    padding: 0;
  }
}
