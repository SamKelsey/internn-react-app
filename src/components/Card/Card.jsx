import React, { Component } from "react";
import "./card.scss";

class Card extends Component {
  render() {
    const { icon, title, body, height } = this.props;

    return (
      <div className="about-card" style={{ height: height }}>
        <i className={icon}></i>
        <h2>{title}</h2>
        <p>{body}</p>
      </div>
    );
  }
}

export default Card;
