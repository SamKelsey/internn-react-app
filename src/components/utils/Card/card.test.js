import Card from "./Card";
import ShallowRenderer from "react-test-renderer/shallow";

const renderer = new ShallowRenderer();

test("renders correctly", () => {
  renderer.render(
    <Card icon="fas fa-users" title="test title" body="test body" />
  );
  const component = renderer.getRenderOutput();
  expect(component).toMatchSnapshot();
});
