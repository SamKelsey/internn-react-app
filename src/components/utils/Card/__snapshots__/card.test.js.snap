// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="about-card"
  style={
    Object {
      "height": undefined,
    }
  }
>
  <i
    className="fas fa-users"
  />
  <h2>
    test title
  </h2>
  <p>
    test body
  </p>
</div>
`;
